File: Backend/controller/kas/kip17/kip17TokenTransfer.js
const wallet = require('../../../service/kas/wallet');
const kip17 = require('../../../service/kas/kip17');

const kip17TokenTransfer = async function (req, res) {
    try {
        const check = await wallet.kasCheckAccount(req.body.to);

        if (check['statusCode'] != 200) {
            res.status(400);
            res.json({statusCode: 400, msg: "수신자의 주소가 올바르지 않습니다."});
            return;
        }
    } catch (e) {
        console.log(e);
        return;
    }

    try {
        const tokenTransfer = await kip17.kasKip17TokenTransfer(req.body.alias, req.body.token_id, req.body.sender, req.body.owner, req.body.to);

        if (tokenTransfer['statusCode'] == 200) {
            res.status(200);
            res.json({statusCode: 200});
        } else {
            const errorCode = tokenTransfer['data']['code'];
            res.status(tokenTransfer['statusCode']);

            switch (errorCode) {
                case 1000000:
                    res.json({statusCode: tokenTransfer['statusCode'], msg: "송신자와 수신자의 주소가 같습니다."});
                    break;
                case 1104404:
                    res.json({statusCode: tokenTransfer['statusCode'], msg: "해당 토큰을 찾을 수 없습니다."});
                    break;
                default:
                    res.json({statusCode: tokenTransfer['statusCode'], msg: "토큰 전송에 실패했습니다. 고객센터에 문의해 주세요."});
                    break;
            }
        }
    } catch (e) {
        console.error(e);
    }
}

module.exports = kip17TokenTransfer;

File: Etiquette/lib/Providers/KAS/Kip17/kip17_token_transfer.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:Etiquette/Models/Settings.dart';

// 토큰 전송
Future<Map<String, dynamic>> kip17TokenTransfer(String alias, String token_id, String sender, String owner, String to) async {
  const url = "$SERVER_IP/kas/kip17/tokenTransfer";
  try {
    final res = await http.post(Uri.parse(url), body: {
      "alias": alias,
      "token_id": token_id,
      "sender": sender,
      "owner": owner,
      "to": to,
    });
    Map<String, dynamic> data = json.decode(res.body);
    return data;
  } catch (ex) {
    Map<String, dynamic> data = {
      "statusCode": 400,
      "msg": ex.toString()
    };
    return data;
  }
}